modules/nlu/src/backend/application/model-entry/model-entry-repo.ts

```diff
@@ -0,0 +1,74 @@
+import * as sdk from 'botpress/sdk'
+
+/**
+ * A bot can have up to one model and one training per language.
+ * "ready" means the model is currently used for prediction.
+ * "not-ready" means the model is currently being made.
+ * */
+type ModelStateStatus = 'ready' | 'not-ready'
+
+interface ModelEntryPrimaryKey {
+  botId: string
+  language: string
+  status: ModelStateStatus
+}
+
+interface ModelEntryRow extends ModelEntryPrimaryKey {
+  modelId: string
+  definitionHash: string
+}
+
+export class ModelEntryRepository {
+  private _tableName = 'model_entry'
+
+  constructor(private _db: sdk.KnexExtended) {}
+
+  private get table() {
+    return this._db.table<ModelEntryRow>(this._tableName)
+  }
+
+  public async initialize() {
+    await this._db.createTableIfNotExists(this._tableName, table => {
```

Comment:
**question:** should we have a down migration to remove the table if some decide downgrade ?